java/runtime/src/main/java/org/ray/runtime/AbstractWorker.java

```diff
@@ -59,53 +62,37 @@
    */
   private long lastCheckpointTimestamp = 0;
 
-
-  public Worker(AbstractRayRuntime runtime) {
+  AbstractWorker(AbstractRayRuntime runtime) {
     this.runtime = runtime;
+    this.functionManager = runtime.functionManager;
   }
 
-  public UniqueId getCurrentActorId() {
-    return currentActorId;
-  }
+  protected List<NativeRayObject> execute(List<String> rayFunctionInfo,
+      List<NativeRayObject> argsBytes) {
+    TaskSpec taskSpec = workerContext.getCurrentTask();
```

Comment:
just wondering if it is necessary to get the whole taskSpec in java?

Reply:
Getting the whole task spec has a minor issue here. that is, the by-value arguments will be copied twice (`argsBytes` copies data as well).
Also, it'd be clearer to just pass the task spec as a parameter of this function.

Reply:
the cost of copying the by-value args could be non-trivial when args are large, is it possible that we don't copy the whole `taskspec` and pass the by value args as shared_ptrs?

Reply:
Thanks for the reminder as I didn't bring by-value args into mind. I've managed to avoid reading the whole task spec in Java. Will update soon. As for passing the task spec as a parameter, I don't think it's necessary since we can get it from worker context.